{"name": "test_for_users_duplicate", "status": "broken", "statusDetails": {"message": "TypeError: list indices must be integers or slices, not str", "trace": "@allure.feature(\"Create already exist User\")\n    @allure.story(\"Authorized user can not add users information\")\n    @allure.severity(allure.severity_level.BLOCKER)\n    def test_for_users_duplicate():\n    \n        response = helper.post_request(url=POST_URL,\n                                              body={'name':'Abhishek Yadav', 'gender':'Male',\n                                                    'email':'yabhishek@outlook.com', 'status':'Active'})\n    \n        check.response_has_status_code(response, 200)\n        check.response_has_field_with_value(response, 'code', 422)\n        check.response_has_fields(\n            response,\n            'data'\n        )\n>       check.response_has_field_value(response, \"field\", 'email')\n\nresponse   = <Response [200]>\n\ntest.py:60: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nresponse = <Response [200]>, field = 'field', expected_value = 'email'\n\n    def response_has_field_value(response, field, expected_value):\n        with allure.step(f'Check response has one of {expected_value} expected value'):\n            res = json.loads(response.text)\n            actual_value = res['data']\n>           assert actual_value[field] in expected_value, f'''\n                Expected value: {expected_value}\n                Actual value: {actual_value}\n            '''\nE           TypeError: list indices must be integers or slices, not str\n\nactual_value = [{'field': 'email', 'message': 'has already been taken'}]\nexpected_value = 'email'\nfield      = 'field'\nres        = {'code': 422,\n 'data': [{'field': 'email', 'message': 'has already been taken'}],\n 'meta': None}\nresponse   = <Response [200]>\n\n..\\testlib\\check.py:18: TypeError"}, "steps": [{"name": "Send POST request by authorized user to url:\n https://gorest.co.in/public-api/users", "status": "passed", "attachments": [{"source": "ef3807d5-b9db-413d-9301-4df31a68fa9f-attachment.attach"}, {"source": "fae4372f-0c66-4e06-9791-277928b3921b-attachment.attach"}], "start": 1620598246911, "stop": 1620598247362}, {"name": "Check response has one of (200,) status code", "status": "passed", "start": 1620598247363, "stop": 1620598247363}, {"name": "Check response jsondata have code field with 422 value", "status": "passed", "start": 1620598247363, "stop": 1620598247364}, {"name": "Check response jsondata have all of ('data',) fields", "status": "passed", "start": 1620598247364, "stop": 1620598247364}, {"name": "Check response has one of email expected value", "status": "broken", "statusDetails": {"message": "TypeError: list indices must be integers or slices, not str\n", "trace": "  File \"D:\\MyWorkspace\\api_tests_python_allure\\testlib\\check.py\", line 18, in response_has_field_value\n    assert actual_value[field] in expected_value, f'''\n"}, "start": 1620598247364, "stop": 1620598247365}], "start": 1620598246911, "stop": 1620598247367, "uuid": "684230b6-dc4b-477b-bd23-b9509da98283", "historyId": "90db21108ff4a0e23ca7eeedefc30d6d", "testCaseId": "2b61faf00e535626e31545461e44eb04", "fullName": "tests.test#test_for_users_duplicate", "labels": [{"name": "severity", "value": "blocker"}, {"name": "story", "value": "Authorized user can not add users information"}, {"name": "feature", "value": "Create already exist User"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test"}, {"name": "host", "value": "abhishekkyd"}, {"name": "thread", "value": "7668-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test"}]}